src/main/java/org/team1540/robot2024/subsystems/elevator/ElevatorIOTalonFX.java

```diff
@@ -0,0 +1,99 @@
+        TalonFXConfiguration config = new TalonFXConfiguration();
+        config.CurrentLimits.SupplyCurrentLimitEnable = true;
+        config.CurrentLimits.SupplyCurrentLimit = 40.0;
+        config.CurrentLimits.SupplyCurrentThreshold = 60.0;
+        config.CurrentLimits.SupplyTimeThreshold = 0.1;
+        config.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+        config.Feedback.SensorToMechanismRatio = Constants.Elevator.MOTOR_ROTS_TO_METERS;
+        // TODO: this might not actually be inverted, so fix it if neccesary
+        follower.setControl(new Follower(leader.getDeviceID(), true));
+
+        BaseStatusSignal.setUpdateFrequencyForAll(
+            50.0, leaderPosition, leaderVelocity, leaderAppliedVolts, leaderCurrent, followerCurrent, topLimitStatus, bottomLimitStatus);
+        leader.optimizeBusUtilization();
+        follower.optimizeBusUtilization();
+
+        // setting slot 0 gains
+        Slot0Configs slot0Configs = config.Slot0;
+        slot0Configs.kS = Constants.Elevator.KS;
+        slot0Configs.kV = Constants.Elevator.KV;
+        slot0Configs.kA = Constants.Elevator.KA;
+        slot0Configs.kP = Constants.Elevator.KP;
+        slot0Configs.kI = Constants.Elevator.KI;
+        slot0Configs.kD = Constants.Elevator.KD;
+        slot0Configs.kG = Constants.Elevator.KG;
+
+        // setting Motion Magic Settings
+        MotionMagicConfigs motionMagicConfigs = config.MotionMagic;
+        motionMagicConfigs.MotionMagicCruiseVelocity = Constants.Elevator.CRUISE_VELOCITY_MPS;
+        motionMagicConfigs.MotionMagicAcceleration = Constants.Elevator.MAXIMUM_ACCELERATION_MPS2;
+        motionMagicConfigs.MotionMagicJerk = Constants.Elevator.JERK_MP3;
+
+        config.HardwareLimitSwitch.ForwardLimitEnable = true;
+        config.HardwareLimitSwitch.ReverseLimitEnable = true;
+        leader.getConfigurator().apply(config);
+        follower.getConfigurator().apply(config);
+
+    }
+
+    @Override
+    public void updateInputs(ElevatorIOInputs inputs) {
+        BaseStatusSignal.refreshAll(
+                leaderPosition,
+                leaderVelocity,
+                leaderAppliedVolts,
+                leaderCurrent,
+                followerCurrent,
+                topLimitStatus,
+                bottomLimitStatus);
+
+        inputs.positionMeters = leaderPosition.getValue();
+        inputs.velocityRPM = leaderVelocity.getValue();
+        inputs.voltage = leaderAppliedVolts.getValue();
+        inputs.current = new double[] {leaderCurrent.getValue(), followerCurrent.getValue()};
+        inputs.upperLimit = topLimitStatus.getValue() == ForwardLimitValue.Open;
+        inputs.lowerLimit = bottomLimitStatus.getValue() == ReverseLimitValue.Open;
+    }
+
+    @Override
+    public void setPositionMeters(double positionMeters) {
+        leader.setControl(motionMagicOut.withPosition(positionMeters));
+    }
+
+    @Override
+    public void setVoltage(double voltage) {
+        leader.set(voltage*12);
```

Comment:
This should call `leader.setControl` with a `VoltageOut` control request.

Reply:
I'm changing it to `setPercent` as per Zach and Adley's requests.